src/commons/sagas/StoriesSaga.ts

```diff
@@ -33,11 +35,34 @@ import { safeTakeEvery as takeEvery } from './SafeEffects';
 export function* storiesSaga(): SagaIterator {
   yield takeLatest(GET_STORIES_LIST, function* () {
     const tokens: Tokens = yield selectTokens();
-    const allStories: StoryListView[] = yield call(async () => {
-      const resp = await getStories(tokens);
+
+    const draftStories: StoryListView[] = yield call(async () => {
+      const resp = await getStories(tokens, StoryStatus.Draft);
+      return resp ?? [];
+    });
+
+    const pendingStories: StoryListView[] = yield call(async () => {
+      const resp = await getStories(tokens, StoryStatus.Pending);
+      return resp ?? [];
+    });
+
+    const rejectedStories: StoryListView[] = yield call(async () => {
+      const resp = await getStories(tokens, StoryStatus.Rejected);
+      return resp ?? [];
+    });
+
+    const publishedStories: StoryListView[] = yield call(async () => {
+      const resp = await getStories(tokens, StoryStatus.Published);
       return resp ?? [];
     });
 
+    const allStories: StoryListViews = {
+      draft: draftStories,
+      pending: pendingStories,
+      rejected: rejectedStories,
+      published: publishedStories
+    };
```

Comment:
This design doesn't look right to me. To the user/the frontend, the "listing" of stories should be transparent.

In other words, the FE just makes one "list stories" query, and the backend returns a list of stories that the user has access to. The permissions/different types of stories should be transparent and the frontend can subsequently later filter the story by the status. You are adding a status field inside the story model in the backend but it's not being utilised here at all.